core/src/main/java/tc/oc/pgm/listeners/VanishListener.java

```diff
@@ -0,0 +1,53 @@
+package tc.oc.pgm.listeners;
+
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import tc.oc.pgm.api.event.PlayerVanishEvent;
+import tc.oc.pgm.api.integration.Integration;
+import tc.oc.pgm.api.party.Competitor;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.MatchPlayerAddEvent;
+import tc.oc.pgm.listeners.JoinLeaveAnnouncer.JoinVisibility;
+
+public class VanishListener implements Listener {
+
+  @EventHandler(priority = EventPriority.NORMAL)
+  public void setVanishState(PlayerVanishEvent event) {
+    MatchPlayer player = event.getPlayer();
+
+    // Ensure player is an observer
+    event.getMatch().setParty(player, event.getMatch().getDefaultParty());
```

Comment:
This is a tricky one, but you may not be allowed to join obs (eg: death tracker says you're in danger) which means you'll vanish while on blue/red team.

Could make it so if you do not change party it doesn't set vanished, but the message that you left has already been sent.

Making this switch before sending the message is also not possible because then you'd always appear as in obs in the leave message when vanishing.

It's a tricky one

Reply:
Had not thought of this! Good observation. 

Indeed, very tricky.

Could we perhaps modify `CombatLogTracker#getImminentDeath` so we can call that from `VanishCommand` and just do a quick check before vanishing? 

Or an alternative, maybe split up vanishing into multiple events. `PrePlayerVanishEvent` Allow for us to cancel the vanish before the leave messages are shown.